src/core/abr/representation_estimator.ts

```diff
@@ -237,6 +245,9 @@ export default function RepresentationEstimator({
 
   const currentRepresentation$ = bufferEvents$.pipe(
     filter((e) : e is IABRRepresentationChange => e.type === "representationChange"),
+    tap(() => {
+      shouldIgnoreMetrics = generateCachedSegmentDetector(bandwidthEstimator);
```

Comment:
:+1: I found it to be much more maintainable and readable, that's cool!

Just why do you reset it at each representation change?